pahfit/model.py

```diff
@@ -284,9 +284,19 @@ def amp_guess(row, fwhm):
         else:
             loop_over_non_fixed("line", "power", lambda row: some_flux)
 
-        # set the fwhms in the features table
+        # Set the fwhms in the features table. Slightly different logic,
+        # as the fwhm for lines are masked by default
         if calc_line_fwhm:
-            loop_over_non_fixed("line", "fwhm", line_fwhm_guess, force=True)
+            for row_index in np.where(self.features["kind"] == "line")[0]:
+                row = self.features[row_index]
+                if np.ma.is_masked(row["fwhm"]):
+                    self.features[row_index]["fwhm"] = (
+                        line_fwhm_guess(row),
```

Comment:
Uh, so now we _are_ sanitizing the `Features` table, the very one the user can see and operate on.  Were we going to leave it alone and unsullied by instrument artifacts?  Why couldn't this be done right when you feed a feature to `Fitter`?

Reply:
Leaving it alone would definitely be best. I think it was a bit of a hack to let the `guess()` step set the fwhm. I will give it some thought, and see if I can make the main use cases work without guessing `fwhm` for the lines in this step. Then we just need to make sure that the right fwhm values are generated on the fly, e.g. when plotting etc, but also that we don't accidentally overwrite any fwhm values that were fit.

Reply:
I agree: leave it alone.  If `Features` is ever and always fully and completely divorced from any aspect of instrument information, `plot` will need to "just in time" marry the instrument information from the spectrum(/a) it is plotting.  As will `fit` (modulo "re-use") and `tabulate`.  And (in another way) `guess`.  But all of them get it "just in time" (for convenience we can calculate and cache it perhaps).  `Model` probably needs a function to do this.

For lines, the simplest solution, and one I'm advocating now, is to treat line FWHM as a fixed **instrument exclusive**.  None of your business.  See #293.  `guess` will need to "know about it", since it has to guess power from a flux, but we have that information from the spectra being "guessed".  Line FWHM never appears in the `Features` table.  Setting it on lines there does nothing, just as it does nothing for starlight.  

The real advantage of this _pure science_ `Features` approach is that _any_ spectra or combination of spectra can be used to update `Features`.  If you choose to plot or re-fit it using a very different flavor of spectra, so be it, you probably have your reasons.  We will "do the right thing" with the instrument info you gave us.

As for stitched spectra, `instrument` already provides an intermediate resolution, which should be good enough for most types of stitching.  Users can always _cheat_ by opening up `sigma_v` on lines, but they will then know they are cheating.

I recognize this could be a significant change, so for now let's stick to the semi-hack line and guess FWHM approach we have.  Then after v2.6 is out and the other fitter is underway, we can update the API, move to remove it, add `sigma_v`, and update docs (for v3).

Reply:
Ok, I added a TODO comment in guess(). We can address this after the merge into dev via #293